### PR TITLE
how-to/deploy: update cmd and toml descriptions for deploy of binlog

### PR DESCRIPTION
**Pull request by lichunzhu**
### What is changed, added or deleted? 
Issue: [TOOL-1462](https://internal.pingcap.net/jira/browse/TOOL-1462)
More descriptions for pump's & drainer's cmd and toml configurations are added in how to deploy binlog.

### Which version does your change affect?
dev, v3.0, v2.1

### Checklist 

- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block

### CODE REVIEW

File: dev/how-to/deploy/tidb-binlog.md [hunk 1]
```diff
@@ -426,10 +463,16 @@ Drainer="192.168.0.13"
         # 顺序依次还原成单个事务进行同步（下游服务类型为 MySQL, 该项设置为 False）
         disable-dispatch = false
 
+        # safe mode 设置为 true 时开启安全模式, 将 update 语句拆分为 delete + replace 语句
```
Comment (july2993) [inline dev/how-to/deploy/tidb-binlog.md]: safe mode 会使写下游 MySQL/TiDB 可重入,  会用 replace 跟 delete + replace 替换 insert 跟 update 语句。

File: dev/how-to/deploy/tidb-binlog.md [hunk 2]
```diff
@@ -353,6 +382,8 @@ Drainer="192.168.0.13"
             Drainer 提供服务的地址(-addr="192.168.0.13:8249")
         -c int
             同步下游的并发数，该值设置越高同步的吞吐性能越好 (default 1)
+        -cache-binlog-count int
+            缓存中的binlog数目限制(默认 65536)
```
Comment (suzaku) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
            缓存中的 binlog 数目限制(默认 65536)
```

**Comment by july2993** [comment]
looks need update v2.1 and v3.0 too now

**Comment by lichunzhu** [comment]
> looks need update v2.1 and v3.0 too now

I will work on it.

File: dev/how-to/deploy/tidb-binlog.md [hunk 3]
```diff
@@ -299,6 +299,8 @@ Drainer="192.168.0.13"
             Prometheus Pushgateway 地址，不设置则禁止上报监控信息
         -metrics-interval int
             监控信息上报频率 (默认 15，单位 秒)
+        -node-id string
+            pump 节点的唯一识别ID，如果不指定程序会根据主机名和监听端口自动生成
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
            Pump 节点的唯一识别 ID，如果不指定，程序会根据主机名和监听端口自动生成
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 4]
```diff
@@ -326,9 +328,36 @@ Drainer="192.168.0.13"
         # PD 集群节点的地址
         pd-urls = "http://192.168.0.16:2379,http://192.168.0.15:2379,http://192.168.0.14:2379"
 
+        # [security]
+        # 如无特殊安全设置需要，该部分一般都注解掉
+        # 包含与集群连接的受信任SSL CA列表的文件路径
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 包含与集群连接的受信任 SSL CA 列表的文件路径
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 5]
```diff
@@ -326,9 +328,36 @@ Drainer="192.168.0.13"
...
+        # 包含与集群连接的 PEM 形式的 X509 certificate 的路径
+        # ssl-cert = "/path/to/drainer.pem"
+        # 包含与集群链接的 PEM 形式的 X509 key 的路径
+        # ssl-key = "/path/to/drainer-key.pem"
+
         # [storage]
         # 设置为 true（默认值）来保证可靠性，确保 binlog 数据刷新到磁盘
         # sync-log = true
+
+        # 当可用磁盘容量小于该设置值时 Pump 将停止写入数据
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 6]
```diff
@@ -326,9 +328,36 @@ Drainer="192.168.0.13"
...
+        # 42 MB -> 42000000, 42 mib -> 44040192
+        # default: 10 gib
+        # stop-write-at-available-space = "10 gib"
+
+        # Pump 内嵌的 LSM DB 设置, 除非对该部分很了解否则一般注解掉
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # Pump 内嵌的 LSM DB 设置，除非对该部分很了解，否则一般注解掉
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 7]
```diff
@@ -353,6 +382,8 @@ Drainer="192.168.0.13"
             Drainer 提供服务的地址(-addr="192.168.0.13:8249")
         -c int
             同步下游的并发数，该值设置越高同步的吞吐性能越好 (default 1)
+        -cache-binlog-count int
+            缓存中的 binlog 数目限制(默认 65536)
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
            缓存中的 binlog 数目限制（默认 65536）
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 8]
```diff
@@ -380,10 +411,13 @@ Drainer="192.168.0.13"
             Prometheus Pushgateway 地址，不设置则禁止上报监控信息
         -metrics-interval int
             监控信息上报频率（默认 15，单位：秒）
+        -node-id string
+            drainer 节点的唯一识别ID，如果不指定程序会根据主机名和监听端口自动生成
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
            drainer 节点的唯一识别 ID，如果不指定，程序会根据主机名和监听端口自动生成
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 9]
```diff
@@ -380,10 +411,13 @@ Drainer="192.168.0.13"
...
         -pd-urls string
             PD 集群节点的地址 (-pd-urls="http://192.168.0.16:2379,http://192.168.0.15:2379,http://192.168.0.14:2379")
         -safe-mode
-            是否开启安全模式（将 update 语句拆分为 delete + replace 语句）
+            是否开启安全模式使得下游 MySQL/TiDB 可重入
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]: How to understand "可重入"? Could you make it easier to understand?

File: dev/how-to/deploy/tidb-binlog.md [hunk 10]
```diff
@@ -426,10 +460,17 @@ Drainer="192.168.0.13"
         # 顺序依次还原成单个事务进行同步（下游服务类型为 MySQL, 该项设置为 False）
         disable-dispatch = false
 
+        # safe mode 会使写下游 MySQL/TiDB 可重入
+        # 会用 replace 替换 insert 语句, 用 delete + replace 替换 update 语句
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 会用 replace 替换 insert 语句，用 delete + replace 替换 update 语句
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 11]
```diff
@@ -426,10 +460,17 @@ Drainer="192.168.0.13"
         # 顺序依次还原成单个事务进行同步（下游服务类型为 MySQL, 该项设置为 False）
         disable-dispatch = false
 
+        # safe mode 会使写下游 MySQL/TiDB 可重入
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]: ditto

File: dev/how-to/deploy/tidb-binlog.md [hunk 12]
```diff
@@ -426,10 +460,17 @@ Drainer="192.168.0.13"
...
+        # 会用 replace 替换 insert 语句, 用 delete + replace 替换 update 语句
+        safe-mode = false
+
         # Drainer 下游服务类型（默认为 mysql）
         # 参数有效值为 "mysql"，"file"，"kafka"，"flash"
         db-type = "mysql"
 
+        # commit ts 在该列表中的事务将被过滤，不会同步至下游
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]: This sentence is also a little bit confusing. Please make it clearer. Like what's the relationship between "commit ts" and "该列表"?

File: dev/how-to/deploy/tidb-binlog.md [hunk 13]
```diff
@@ -465,13 +506,19 @@ Drainer="192.168.0.13"
...
         # kafka-addrs = "127.0.0.1:9092"
         # kafka-version = "0.8.2.0"
+        # kafka-max-messages = 1024
 
         # 保存 binlog 数据的 Kafka 集群的 topic 名称，默认值为 <cluster-id>_obinlog
         # 如果运行多个 Drainer 同步数据到同一个 Kafka 集群，每个 Drainer 的 topic-name 需要设置不同的名称
         # topic-name = ""
+
+        [syncer.to.checkpoint]
+        # 当下游是 mysql 或 tidb 时可以开启该选项，以改变保存 checkpoint 的数据库
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 当下游是 MySQL 或 TiDB 时可以开启该选项，以改变保存 checkpoint 的数据库
```

File: dev/how-to/deploy/tidb-binlog.md [hunk 14]
```diff
@@ -465,13 +506,19 @@ Drainer="192.168.0.13"
 
         # db-type 设置为 kafka 时，Kafka 相关配置
         # [syncer.to]
+        # kafka 和 zookeeper-addrs 只需要一个，只给 zookeeper-addrs 时程序会优先用 zookeeper 中的 kafka 地址
```
Comment (lilin90) [inline dev/how-to/deploy/tidb-binlog.md]:
```suggestion
        # kafka-addrs 和 zookeeper-addrs 只需要一个，只设置 zookeeper-addrs 时，程序会优先用 zookeeper 中的 kafka 地址
```

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 15]
```diff
@@ -427,10 +461,17 @@ Drainer="192.168.0.13"
...
 
+        # safe mode 会使写下游 MySQL/TiDB 可被重复写入，v3.0.1后支持该功能
+        # 会用 replace 替换 insert 语句，用 delete + replace 替换 update 语句
+        safe-mode = false
+
         # Drainer 下游服务类型（默认为 mysql）
         # 参数有效值为 "mysql"，"file"，"kafka"，"flash"
         db-type = "mysql"
 
+        # 事务的 commit ts 若在该列表中，则该事务将被过滤，不会同步至下游
```
Comment (july2993) [inline v3.0/how-to/deploy/tidb-binlog.md]: add start support version

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 16]
```diff
@@ -427,10 +461,17 @@ Drainer="192.168.0.13"
         # 顺序依次还原成单个事务进行同步（下游服务类型为 MySQL, 该项设置为 False）
         disable-dispatch = false
 
+        # safe mode 会使写下游 MySQL/TiDB 可被重复写入，v3.0.1后支持该功能
```
Comment (july2993) [inline v3.0/how-to/deploy/tidb-binlog.md]: no need "v3.0.1后支持该功能" all 3.* support

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 17]
```diff
@@ -381,10 +412,13 @@ Drainer="192.168.0.13"
             Prometheus Pushgateway 地址，不设置则禁止上报监控信息
         -metrics-interval int
             监控信息上报频率（默认 15，单位：秒）
+        -node-id string
+            v3.0.2后支持该功能，drainer 节点的唯一识别 ID，如果不指定程序会根据主机名和监听端口自动生成
         -pd-urls string
             PD 集群节点的地址 (-pd-urls="http://192.168.0.16:2379,http://192.168.0.15:2379,http://192.168.0.14:2379")
         -safe-mode
-            是否开启安全模式（将 update 语句拆分为 delete + replace 语句）
+            v3.0.1后支持该功能，是否开启安全模式使得下游 MySQL/TiDB 可被重复写入
```
Comment (july2993) [inline v3.0/how-to/deploy/tidb-binlog.md]: ditto

File: v2.1/how-to/deploy/tidb-binlog.md [hunk 18]
```diff
@@ -426,6 +458,10 @@ Drainer="192.168.0.13"
         # 顺序依次还原成单个事务进行同步（下游服务类型为 MySQL, 该项设置为 False）
         disable-dispatch = false
 
+        # safe mode 会使写下游 MySQL/TiDB 可被重复写入，v2.1.14后支持该功能
```
Comment (july2993) [inline v2.1/how-to/deploy/tidb-binlog.md]: no neeed "v2.1.14后支持该功能" all version support

File: v2.1/how-to/deploy/tidb-binlog.md [hunk 19]
```diff
@@ -383,7 +414,8 @@ Drainer="192.168.0.13"
         -pd-urls string
             PD 集群节点的地址 (-pd-urls="http://192.168.0.16:2379,http://192.168.0.15:2379,http://192.168.0.14:2379")
         -safe-mode
-            是否开启安全模式（将 update 语句拆分为 delete + replace 语句）
+            v2.1.14后支持该功能，是否开启安全模式使得下游 MySQL/TiDB 可被重复写入
```
Comment (july2993) [inline v2.1/how-to/deploy/tidb-binlog.md]: ditto

**Comment by lichunzhu** [comment]
@july2993 PTAL

File: v2.1/how-to/deploy/tidb-binlog.md [hunk 20]
```diff
@@ -326,9 +328,36 @@ Drainer="192.168.0.13"
...
+        # ssl-key = "/path/to/drainer-key.pem"
+
         # [storage]
         # 设置为 true（默认值）来保证可靠性，确保 binlog 数据刷新到磁盘
         # sync-log = true
+
+        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据
+        # 42 MB -> 42000000, 42 mib -> 44040192
+        # default: 10 gib
+        # stop-write-at-available-space = "10 gib"
```
Comment (july2993) [inline v2.1/how-to/deploy/tidb-binlog.md]: add support version

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 21]
```diff
@@ -327,9 +329,36 @@ Drainer="192.168.0.13"
...
+        # ssl-key = "/path/to/drainer-key.pem"
+
         # [storage]
         # 设置为 true（默认值）来保证可靠性，确保 binlog 数据刷新到磁盘
         # sync-log = true
+
+        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据
+        # 42 MB -> 42000000, 42 mib -> 44040192
+        # default: 10 gib
+        # stop-write-at-available-space = "10 gib"
```
Comment (july2993) [inline v3.0/how-to/deploy/tidb-binlog.md]: add support version

File: v2.1/how-to/deploy/tidb-binlog.md [hunk 22]
```diff
@@ -326,9 +328,36 @@ Drainer="192.168.0.13"
...
+        # 包含与集群连接的 PEM 形式的 X509 certificate 的路径
+        # ssl-cert = "/path/to/drainer.pem"
+        # 包含与集群链接的 PEM 形式的 X509 key 的路径
+        # ssl-key = "/path/to/drainer-key.pem"
+
         # [storage]
         # 设置为 true（默认值）来保证可靠性，确保 binlog 数据刷新到磁盘
         # sync-log = true
+
+        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据，v2.1.14后支持该功能
```
Comment (lilin90) [inline v2.1/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据，v2.1.14 后支持该功能
```

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 23]
```diff
@@ -327,9 +329,36 @@ Drainer="192.168.0.13"
...
+        # 包含与集群连接的 PEM 形式的 X509 certificate 的路径
+        # ssl-cert = "/path/to/drainer.pem"
+        # 包含与集群链接的 PEM 形式的 X509 key 的路径
+        # ssl-key = "/path/to/drainer-key.pem"
+
         # [storage]
         # 设置为 true（默认值）来保证可靠性，确保 binlog 数据刷新到磁盘
         # sync-log = true
+
+        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据，v3.0.1后支持该功能
```
Comment (lilin90) [inline v3.0/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 当可用磁盘容量小于该设置值时，Pump 将停止写入数据，v3.0.1 后支持该功能
```

File: v3.0/how-to/deploy/tidb-binlog.md [hunk 24]
```diff
@@ -427,10 +461,17 @@ Drainer="192.168.0.13"
...
 
+        # safe mode 会使写下游 MySQL/TiDB 可被重复写入
+        # 会用 replace 替换 insert 语句，用 delete + replace 替换 update 语句
+        safe-mode = false
+
         # Drainer 下游服务类型（默认为 mysql）
         # 参数有效值为 "mysql"，"file"，"kafka"，"flash"
         db-type = "mysql"
 
+        # 事务的 commit ts 若在该列表中，则该事务将被过滤，不会同步至下游，v3.0.2后支持该功能
```
Comment (lilin90) [inline v3.0/how-to/deploy/tidb-binlog.md]:
```suggestion
        # 事务的 commit ts 若在该列表中，则该事务将被过滤，不会同步至下游，v3.0.2 后支持该功能
```

**Pull request merged**